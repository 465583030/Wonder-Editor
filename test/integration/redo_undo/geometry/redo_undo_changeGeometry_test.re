open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: change geometry", () => {
    let sandbox = getSandboxDefaultVal();

    let _getFromArray = (array, index) => ArrayService.getNth(index, array);

    beforeEach(() => {
      sandbox := createSandbox();

      MainEditorSceneTool.initState(~sandbox, ());

      EventListenerTool.buildFakeDom()
      |> EventListenerTool.stubGetElementByIdReturnFakeDom;
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));
    let _simulateChangeGeometry = () => {
      let component =
        BuildComponentTool.buildGeometry(
          TestTool.buildEmptyAppState(),
          GameObjectTool.getCurrentGameObjectGeometry(),
        );

      BaseEventTool.triggerComponentEvent(
        component,
        MainEditorGeometryTool.triggerClickShowGeometryGroup,
      );

      BaseEventTool.triggerComponentEvent(
        component,
        MainEditorGeometryTool.getSphereDomIndex()
        |> MainEditorGeometryTool.triggerClickSpecificGeometry,
      );
    };

    let _beforeEach = () =>
      MainEditorSceneTool.createDefaultScene(
        sandbox,
        MainEditorSceneTool.setFirstBoxTobeCurrentSceneTreeNode,
      );
    let _afterEach = () => ();

    RedoUndoTool.testRedoUndoOneStep(
      sandbox,
      "prepare first step: set currentSceneTreeNode",
      (_simulateChangeGeometry, _beforeEach, _afterEach),
      BuildComponentForCurryTool.buildGeometry,
    );
  });