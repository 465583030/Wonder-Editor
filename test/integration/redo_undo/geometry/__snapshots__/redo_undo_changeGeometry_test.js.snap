// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: change geometry prepare first step: set currentSceneTreeNode test redo operate test redo one step undo step which from first to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-inspector-geometry"
>
  <div
    className="geometry-select"
  >
    Sphere
    <span
      className="select-title"
      onClick={[Function]}
    >
      select
    </span>
  </div>
</article>
`;

exports[`redo_undo: change geometry prepare first step: set currentSceneTreeNode test undo operate test not undo 1`] = `
<article
  className="wonder-inspector-geometry"
>
  <div
    className="geometry-select"
  >
    Sphere
    <span
      className="select-title"
      onClick={[Function]}
    >
      select
    </span>
  </div>
</article>
`;

exports[`redo_undo: change geometry prepare first step: set currentSceneTreeNode test undo operate test undo one step undo step which from first to zero 1`] = `
<article
  className="wonder-inspector-geometry"
>
  <div
    className="geometry-select"
  >
    Cube
    <span
      className="select-title"
      onClick={[Function]}
    >
      select
    </span>
  </div>
</article>
`;
